connexion/operation.py

```diff
@@ -123,44 +123,67 @@ def validate_defaults(self):
 
     def resolve_reference(self, schema):
         schema = deepcopy(schema)  # avoid changing the original schema
-        # find the object we need to resolve/update
+        self.check_references(schema)
+
+        # find the object we need to resolve/update if this is not a proper SchemaObject
+        # e.g a response or parameter object
         for obj in schema, schema.get('items'):
             reference = obj and obj.get('$ref')  # type: str
             if reference:
                 break
         if reference:
-            if not reference.startswith('#/'):
-                raise InvalidSpecification(
-                    "{method} {path}  '$ref' needs to start with '#/'".format(**vars(self)))
-            path = reference.split('/')
-            definition_type = path[1]
-            try:
-                definitions = self.definitions_map[definition_type]
-            except KeyError:
-                raise InvalidSpecification(
-                    "{method} {path}  '$ref' needs to point to definitions or parameters".format(**vars(self)))
-            definition_name = path[-1]
-            try:
-                # Get sub definition
-                definition = deepcopy(definitions[definition_name])
-            except KeyError:
-                raise InvalidSpecification("{method} {path} Definition '{definition_name}' not found".format(
-                    definition_name=definition_name, method=self.method, path=self.path))
-
-            # resolve object properties too
-            for prop, prop_spec in definition.get('properties', {}).items():
-                resolved = self.resolve_reference(prop_spec.get('schema', {}))
-                if not resolved:
-                    resolved = self.resolve_reference(prop_spec)
-
-                if resolved:
-                    definition['properties'][prop] = resolved
-
+            definition = deepcopy(self._retrieve_reference(reference))
             # Update schema
             obj.update(definition)
             del obj['$ref']
+
+        # if there is a schema object on this param or response, then we just
+        # need to include the defs and it can be validated by jsonschema
+        if 'schema' in schema:
+            schema['schema']['definitions'] = self.definitions
+            return schema
+
         return schema
 
+    def check_references(self, schema):
```

Comment:
Fair point, I can add docs ... I was actually am of a split mind on whether I should include it ... on the one hand it will check most references and thus preserve the current behavior of failing on load ... on the other hand, jsonschema will check refs when it validates, and will give a somewhat more meaningful message.